app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,180 @@
+
+  end
+
+  def unshipped
+    @merchant = Merchant.find(params[:merchant_id])
+    @orders = @merchant.orders.where(status: "complete").uniq.reverse
+    @order_items = find_relevant_order_items(@orders, @merchant.id)
+    @order_items_unshipped = shipped?(@order_items, false)
+
+    @unshipped_orders = find_orders_by_status(@orders, false, @merchant.id)
+
+    @shipped_revenue = revenue(@order_items, true)
+    @unshipped_revenue = revenue(@order_items, false)
+    @shipped_count = count_items(@order_items, true)
+    @unshipped_count = count_items(@order_items, false)
+  end
+
+  def show
+    # a merchant can view a particular order and all of its details (i.e. order_items/totals, etc.)
+    @order = Order.find(params[:id])
+
+    @order_items = Merchant.find(params[:merchant_id]).order_items
+
+    @redacted_cc = redacted_cc(@order.credit_card)
+  end
+
+  def edit
+    # every time a new order_item is added/removed from the cart
+    # when the customer adds their payment details
+    # params[:id] is the order.id 
+    if session[:order_id] == params[:id].to_i
+      @order = Order.find(params[:id])
+    else
+      flash[:error] = "You do not have access to that order."
+
+      redirect_to root_path
+    end
+  end
+
+  def update
+    @order = Order.find(params[:id])
+    @order.update(order_params)
+    @order.update(status: "complete")
+
+    @order.order_items.each do |order_item|
+      product = Product.find(order_item.product_id)
+      previous_stock = product[:stock]
+      ordered_stock = order_item.quantity
+
+      new_stock = previous_stock.to_i - ordered_stock.to_i
+      product.update(stock: new_stock)
+    end
+
+    session[:order_id] = nil # this clears the cart after you've checked out
+
+    redirect_to order_confirmation_path(params[:id])
+  end
+
+  def confirmation
+    @order = Order.find(params[:order_id])
+    @order_items = @order.order_items
+    @total = get_total(@order_items)
+    @customer_info = get_customer_info(@order)
+  end
+
+  def get_total(order_items)
+    order_items.inject(0) do |sum, i|
+      sum + i.revenue
+    end
+  end
+
+  def get_customer_info(order)
+    customer_info = []
+
+    customer_info.push(order.name, order.email, order.street, order.city, order.state, order.zip)
+  end
+
+  def shipped?(order_items, bool)
+    final_items = []
+    order_items.each do |order_item|
+      if order_item.shipped == bool
+        final_items.push(order_item)
+      end
+    end
+    return final_items
+  end
+
+  def count_items(order_items, bool)
+    count = 0
+    order_items.each do |order_item|
+      if order_item.shipped == bool
+        count += order_item.quantity
+      end
+    end
+    return count
+  end
+
+  def find_relevant_order_items(orders, merchant_id)
+    relevant_order_items = []
+    orders.each do |order|
+      all_order_items = order.order_items
+      order_items = all_order_items.where(merchant_id: merchant_id)
+      relevant_order_items.push(order_items)
```

Comment:
I think that the `+` method could be used instead of `push` and then the `flatten` call on line 143 below would not be needed.